src/util-decode-mime.c

```diff
@@ -1024,88 +1022,107 @@ static int FindUrlStrings(const uint8_t *line, uint32_t len,
+                            PrintChars(SC_LOG_DEBUG, "EXE URL", tempUrl, tempUrlLen);
+                        }
 
-                    PrintChars(SC_LOG_DEBUG, "EXE URL", tempUrl, tempUrlLen);
-                } else {
-                    /* Not an EXE URL */
-                    /* Cut off length at first '/' */
-                    /* If seems that BAESystems had done the following
-                       in support of PEScan.  We don't want it for logging.
-                       Therefore its been removed.
-                    tok = FindString(tempUrl, tempUrlLen, "/");
-                    if (tok != NULL) {
-                        tempUrlLen = tok - tempUrl;
-                    }
-                    */
-                }
+                        /* Make sure remaining URL exists */
+                        if (tempUrlLen > 0) {
+                            if (!(FindExistingUrl(entity, tempUrl, tempUrlLen))) {
+                                /* Now look for numeric IP */
+                                if (IsIpv4Host(urlHost, urlHostLen)) {
+                                    flags |= URL_IS_IP4;
 
-                /* Make sure remaining URL exists */
-                if (tempUrlLen > 0) {
-                    if (!(FindExistingUrl(entity, tempUrl, tempUrlLen))) {
-                        /* Now look for numeric IP */
-                        if (IsIpv4Host(tempUrl, tempUrlLen)) {
-                            flags |= URL_IS_IP4;
+                                    PrintChars(SC_LOG_DEBUG, "IP URL4", tempUrl, tempUrlLen);
+                                } else if (IsIpv6Host(urlHost, urlHostLen)) {
+                                    flags |= URL_IS_IP6;
 
-                            PrintChars(SC_LOG_DEBUG, "IP URL4", tempUrl, tempUrlLen);
-                        } else if (IsIpv6Host(tempUrl, tempUrlLen)) {
-                            flags |= URL_IS_IP6;
+                                    PrintChars(SC_LOG_DEBUG, "IP URL6", tempUrl, tempUrlLen);
+                                }
 
-                            PrintChars(SC_LOG_DEBUG, "IP URL6", tempUrl, tempUrlLen);
+                                /* Add URL list item */
+                                MimeDecAddUrl(entity, tempUrl, tempUrlLen, flags);
+                            } else {
+                                SCFree(tempUrl);
+                            }
+                        } else {
+                            SCFree(tempUrl);
                         }
 
-                        /* Add URL list item */
-                        MimeDecAddUrl(entity, tempUrl, tempUrlLen, flags);
-                    } else {
-                        SCFree(tempUrl);
+                        /* Reset flags for next URL */
+                        flags = 0;
                     }
-                } else {
-                    SCFree(tempUrl);
                 }
-            }
+            } while (fptr != NULL);
         }
-    } while (fptr != NULL);
+    } else {
```

Comment:
I think if we test for this `else` condition first and just return `MIME_DEC_ERR_DATA` directly, we can avoid the above extra layer of indent?